src/app/dashboard/components/filters/form.tsx

```diff
@@ -1,47 +1,108 @@
+import { useDebounce } from 'use-debounce';
+import { useEffect } from 'react';
+
+interface Props {
+  setStartDateHandle: (childData: Dayjs) => void;
+  setEndDateHandle: (childData: Dayjs) => void;
+  setAggregateHandle: (aggregate: number) => void;
+  startDate: Dayjs;
+  endDate: Dayjs;
+  aggregate: number;
+}
+
+const StockChartFiltersForm = (props: Props) => {
+  const [query, setQuery] = useQueryParams({
+    date_from: StringParam,
+    date_to: StringParam,
+    aggregate: NumberParam,
+  });
+  query;
 
-const StockChartFiltersForm: React.FC = () => {
   const methods = useForm<FiltersForm>({
-    defaultValues: getFilterFormDefaultValues(),
+    defaultValues: {
+      startDate: props.startDate,
+      endDate: props.endDate,
+      aggregate: props.aggregate,
+    },
     resolver: zodResolver(FiltersFormSchema),
   });
-  const { handleSubmit, control } = methods;
+  const { handleSubmit, control, setValue } = methods;
 
   const startDate = useWatch({ control, name: 'startDate' });
   const endDate = useWatch({ control, name: 'endDate' });
+  const aggregate = useWatch({ control, name: 'aggregate' });
+  const [debouncedAggregate] = useDebounce(aggregate, 500);
 
   const onSubmit = handleSubmit((data: FiltersForm) => {
-    console.log(data.startDate.format('YYYY-MM-DD'));
-    console.log(data.endDate.format('YYYY-MM-DD'));
+    data;
   });
 
+  function passStartDateToParent(date: Dayjs | null) {
+    if (date === null) return;
+    setValue('startDate', date);
+
+    props.setStartDateHandle(date);
+    setQuery({ date_from: convertISODatetimeWithTimezoneToISODate(date) });
+  }
+  passStartDateToParent;
+
+  function passEndDateToParent(date: Dayjs | null) {
+    if (date === null) return;
+    setValue('endDate', date);
+
+    props.setEndDateHandle(date);
+    setQuery({ date_to: convertISODatetimeWithTimezoneToISODate(date) });
+  }
+  passEndDateToParent;
+
+  function handleAggregateChange(
+    event: React.ChangeEvent<HTMLTextAreaElement | HTMLInputElement>
+  ) {
+    const newValue = event.target.value;
+    const aggregate = newValue === '' ? 1 : parseInt(newValue);
+    setValue('aggregate', aggregate);
+  }
+
+  useEffect(() => {
+    props.setAggregateHandle(debouncedAggregate);
+    setQuery({ aggregate: debouncedAggregate });
+  }, [debouncedAggregate]);
+
   return (
     <StyledForm methods={methods} onSubmit={onSubmit}>
       <DateInput
         name="startDate"
         label="Start Date"
         minDate={minDateBoundary}
         maxDate={endDate}
+        onChange={passStartDateToParent}
       />
       <DateInput
         name="endDate"
         label="End Date"
         minDate={startDate}
         maxDate={today}
+        onChange={passEndDateToParent}
+      />
+      <NumberInput
+        name={'aggregate'}
+        label={'aggregate'}
+        value={aggregate}
+        onChange={handleAggregateChange}
+        inputProps={{ min: 1, max: 100, step: 1 }}
       />
```

Comment:
`onChange` oraz `value` są obsługiwane przez `react-hook-form`, będziemy się do nich dostawać używając `useWatch`

```suggestion
    <StyledForm methods={methods}>
      <DateInput
        name="startDate"
        label="Start Date"
        minDate={minDateBoundary}
        maxDate={endDate}
      />
      <DateInput
        name="endDate"
        label="End Date"
        minDate={startDate}
        maxDate={today}
      />
      <NumberInput
        name={'aggregate'}
        label={'aggregate'}
        inputProps={{ min: 1, max: 100, step: 1 }}
      />
```